test/Release/MysqlReleaseGatewayTest.php

```diff
@@ -0,0 +1,161 @@
+            'info' => $url
+        );
+        $this->assertEquals( $expectedRelease, $this->connection->fetchAssoc( 'SELECT * FROM `release` WHERE `id` = ?', [ $id ] ) );
+
+        // Teardown
+        $this->deleteAddedRelease( $id );
+    }
+
+    /**
+     * @test
+     */
+    public function givenNoReleases_getReleaseReturnsNull()
+    {
+        $this->assertNull( $this->gateway->getRelease( 'blergh' ) );
+    }
+
+    /**
+     * @test
+     */
+    public function givenRelease_WhenGetReleaseCalledWithWrongId_ReturnsNull()
+    {
+        $id = $this->addRelease( 'Test release 1', 'a helpful url' );
+        $this->assertNull( $this->gateway->getRelease( 'blergh' ) );
+        $this->deleteAddedRelease( $id );
+    }
+
+    /**
+     * @test
+     */
+    public function givenRelease_getReleaseWithCorrectId_ReturnsRelease()
+    {
+        $releaseName = 'Test release 1';
+        $url = 'a helpful url';
+        $id = $this->addRelease( $releaseName, $url );
+
+        $release = $this->gateway->getRelease( $id );
+        $expectedRelease = new Release( $releaseName, $url, new \DateTime() );
+
+        $this->assertEquals( $expectedRelease->getReleaseName(), $release->getReleaseName() );
+        $this->assertEquals( $expectedRelease->getReleaseInfoUrl(), $release->getReleaseInfoUrl() );
+
+        $this->deleteAddedRelease( $id );
+    }
+
+    /**
+     * @test
+     */
+    public function givenNoReleases_getAllReleasesReturnsEmptyArray()
+    {
+        $this->assertEquals( [], $this->gateway->getAllReleases() );
+    }
+
+    /**
+     * @test
+     */
+    public function givenRelease_getAllReleasesReturnsReleaseInArray()
+    {
+        $releaseName = 'Test release 1';
+        $url = 'a helpful url';
+        $id = $this->addRelease( $releaseName, $url );
+
+        $expectedRelease = new Release( $releaseName, $url, new \DateTime() );
+        $releases =  $this->gateway->getAllReleases();
+
+        $this->assertEquals( $expectedRelease->getReleaseName(), $releases[0]->getReleaseName() );
+        $this->assertEquals( $expectedRelease->getReleaseInfoUrl(), $releases[0]->getReleaseInfoUrl() );
+
+        $this->deleteAddedRelease( $id );
+    }
+
+    /**
+     * @test
+     */
+    public function givenMultipleReleases_getAllReleasesReturnsArrayOfReleases()
+    {
+        $releasesToDelete = array(
+              $this->addRelease( 'Test release 1', 'a helpful url' ),
+              $this->addRelease( 'Test release 2', 'another helpful url' ),
+              $this->addRelease( 'Test release 3', 'a third helpful url' )
+        );
+        $releases =  $this->gateway->getAllReleases();
+
+        $this->assertCount( 3, $releases );
+
+        foreach ( $releasesToDelete as $id ){
+            $this->deleteAddedRelease( $id );
+        }
+    }
+
+    /**
+     * @param $id
+     * @throws \Doctrine\DBAL\Exception\InvalidArgumentException
+     */
+    private function deleteAddedRelease( $id )
```

Comment:
Yeah that will probably come much later, I'm just using it to clean up the db post tests at the moment although thinking about it perhaps `@transaction` would have worked much better. I'll test that out in my local and see.